answerking_app/utils/mixins/GenericMixins.py

```diff
@@ -27,6 +27,15 @@ def update(self, request: Request, *args, **kwargs) -> Response:
             return item_exists_check(err)
 
 
+class RetireMixin(UpdateModelMixin):
+    def retire(self, request: Request, *args, **kwargs) -> Response:
+        try:
+            request.data['retired']: bool = True
+            return super().partial_update(request, *args, **kwargs)
+        except ObjectDoesNotExist as err:
+            return item_exists_check(err)
```

Comment:
I should have pointed this out during the previous review but I didn't notice it, why do you do this check and not just return a 404?

Reply:
yes, and even without the error should be handled by DRF anyway I believe